src/NuGet.Core/NuGet.Packaging/PackageCreation/Utility/NuGetFrameworkNameUtility.cs

```diff
@@ -0,0 +1,88 @@
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Runtime.Versioning;
+
+using NuGet.Frameworks;
+
+namespace NuGet.Packaging
+{
+    public static class NuGetFrameworkNameUtility
+    {
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1021:AvoidOutParameters", MessageId = "1#")]
+        public static NuGetFramework ParseFrameworkNameFromFilePath(string filePath, out string effectivePath)
+        {
+            foreach (string knownFolder in PackagingConstants.Folders.Known)
+            {
+                string folderPrefix = knownFolder + System.IO.Path.DirectorySeparatorChar;
+                if (filePath.Length > folderPrefix.Length &&
+                    filePath.StartsWith(folderPrefix, StringComparison.OrdinalIgnoreCase))
+                {
+                    string frameworkPart = filePath.Substring(folderPrefix.Length);
+
+                    try
+                    {
+                        return NuGetFrameworkNameUtility.ParseFrameworkFolderName(
+                            frameworkPart,
+                            strictParsing: knownFolder == PackagingConstants.Folders.Lib,
+                            effectivePath: out effectivePath);
+                    }
+                    catch (ArgumentException)
+                    {
+                        // if the parsing fails, we treat it as if this file
+                        // doesn't have target framework.
+                        effectivePath = frameworkPart;
+                        return null;
+                    }
+                }
+
+            }
+
+            effectivePath = filePath;
+            return null;
+        }
+
+        /// <summary>
+        /// Parses the specified string into FrameworkName object.
+        /// </summary>
+        /// <param name="path">The string to be parse.</param>
+        /// <param name="strictParsing">if set to <c>true</c>, parse the first folder of path even if it is unrecognized framework.</param>
+        /// <param name="effectivePath">returns the path after the parsed target framework</param>
+        /// <returns></returns>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1021:AvoidOutParameters", MessageId = "2#")]
+        public static NuGetFramework ParseFrameworkFolderName(string path, bool strictParsing, out string effectivePath)
+        {
+            // The path for a reference might look like this for assembly foo.dll:
+            // foo.dll
+            // sub\foo.dll
+            // {FrameworkName}{Version}\foo.dll
+            // {FrameworkName}{Version}\sub1\foo.dll
+            // {FrameworkName}{Version}\sub1\sub2\foo.dll
+
+            // Get the target framework string if specified
+            string targetFrameworkString = Path.GetDirectoryName(path).Split(Path.DirectorySeparatorChar).First();
+
+            effectivePath = path;
+
+            if (String.IsNullOrEmpty(targetFrameworkString))
```

Comment:
string.IsNullOrEmpty instead of String.IsNullOrEmpty? [Guideline](https://github.com/NuGet/NuGet.Client/blob/dev/docs/coding-guidelines.md)#11